### PR TITLE
Yeah, i know. it's huge. Update module and 1.19 tests

### PR DESCRIPTION
**Pull request by angelbarrera92**
Hi team!

I know, this PR is huge, but it contains a lot of new stuff:

Main changes of this HUGE pr:

- Add license checks at CI
- Add 1.19.0 testing
- Update all modules (including major changes in Grafana)
- Add kube-proxy-exporter (daemonset to expose metrics working in all kube setups) (suggested by @lnovara )
- Add eks-sm (suggested by @lnovara )
- Add grafana internal dashboard
- Refactored all dashboards: (suggested by @lnovara )
  - Before this PR, all dashboards got deployed with Grafana. This can cause some issues having dashboards without metrics/product
  - With this PR, every product/package has its own serviceMonitor + grafana Dashboard (as configmap).
    - This caused a lot of changes and headaches. In addition, this update provides a tested upgrade path described in the release notes: `docs/releases/v1.10.0.md`

PD: Special thanks to @lnovara rises some issues/ways to improve the module.
PD2: Pipeline passing: https://ci.sighup.io/sighupio/fury-kubernetes-monitoring/900

### CODE REVIEW

File: katalog/kube-proxy-exporter/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# kube-proxy exporter
+
+It is highly recommended to gather metrics from kube-proxy as it is a critical piece of any Kubernetes Cluster.
+Sometimes (especially in managed clusters) it is not possible to configure kube-proxy to expose metrics, this is why
+this package exists. Another reason to run this exporter instead of just exposing metrics from kube-proxy is the
```
Comment (lnovara) [inline katalog/kube-proxy-exporter/README.md:5]: I think that this description got truncated.

File: katalog/kube-proxy-exporter/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        - "--secure-listen-address=0.0.0.0:18443"
+        - "--upstream=http://127.0.0.1:10249/"
+        - "--logtostderr"
+        - "--v=10"
+        ports:
+        - containerPort: 18443
+          name: https
+        resources:
+          limits:
+            cpu: 20m
```
Comment (lnovara) [inline katalog/kube-proxy-exporter/deploy.yml:45]: Do we really want to set CPU limits this low? Honestly, I would avoid setting them at all for this use case.

File: katalog/node-exporter/kustomization.yaml [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
 namespace: monitoring
 
+bases:
+  - dashboards
+
+images:
+  - name: quay.io/prometheus/node-exporter
+    newTag: v1.0.1
+  - name: quay.io/coreos/kube-rbac-proxy
+    newTag: v0.5.0
```
Comment (lnovara) [inline katalog/node-exporter/kustomization.yaml:14]: Can we use the same version as in `kube-proxy-exporter`?

**Review by lnovara — APPROVED** [review]
LGTM, awesome work! :shipit: ❤️ 

Minor non-blocking comments follow:
- I think that GlusterFS and Kong stuff need to be moved to a separate repository (which already exists AFAIK) because they're not shipped by default
- the `kube-proxy-exporter` doesn't fit 100% as a name to me, I would have called it `kube-proxy-metrics` but I do realize I'm just nipicking 😅
- did you have any chance to check if dashboards and alerts from the [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) project got updated as well?

P.S. I do understand the huge effort you put into this PR but I think we should avoid this big PRs since it makes really difficult to do a thorough review even though the PR title really makes up for it! 😄

File: katalog/kube-proxy-exporter/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        k8s-app: kube-proxy-exporter
+        k8s-proxied-app: kube-proxy
+    spec:
+      tolerations:
+      - operator: Exists
+      serviceAccountName: kube-proxy-exporter
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 65534
+      hostNetwork: true
```
Comment (lnovara) [inline katalog/kube-proxy-exporter/deploy.yml:30]: Why do we need `hostNetwork: true` and `hostPID: true` to expose kube-proxy metrics?

**Comment by angelbarrera92** [comment]
> LGTM, awesome work! :shipit: ❤️
> 
> Minor non-blocking comments follow:
> 
>     * I think that GlusterFS and Kong stuff need to be moved to a separate repository (which already exists AFAIK) because they're not shipped by default
 
I was thinking about it. I don't know what to do. Maybe I can delete it, and if someone needs the dashboards, they are available in the previous releases of this module.

>     * the `kube-proxy-exporter` doesn't fit 100% as a name to me, I would have called it `kube-proxy-metrics` but I do realize I'm just nipicking 😅

Renaming, I think it makes more sense kube-proxy-metrics. my fault!

>     * did you have any chance to check if dashboards and alerts from the [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) project got updated as well?

Nop, I'll take a look. If there is something really interesting, I'll include it.
> 
> P.S. I do understand the huge effort you put into this PR but I think we should avoid this big PRs since it makes really difficult to do a thorough review even though the PR title really makes up for it! 😄

Ye... I know, it was a huge PR because of various reasons... The next release should be smaller.

File: katalog/kube-proxy-exporter/deploy.yml (hunk 4 above)
Comment (angelbarrera92) [reply katalog/kube-proxy-exporter/deploy.yml:30]: We need `hostNetwork` (not sure if we need hostPID) to proxy kube-proxy metrics. As we are not deploying it as a sidecar container we need to access at the host level.

I'll try to test it without `hostPID`

Comment (lnovara) [reply katalog/kube-proxy-exporter/deploy.yml:30]: You're totally right about `hostNetwork` but I don't think `hostPID` should be necessary.

File: katalog/node-exporter/kustomization.yaml (hunk 3 above)
Comment (angelbarrera92) [reply katalog/node-exporter/kustomization.yaml:14]: I'll take a look but we are not using the same image.

File: katalog/kube-proxy-exporter/deploy.yml (hunk 2 above)
Comment (angelbarrera92) [reply katalog/kube-proxy-exporter/deploy.yml:45]: I don't like to put CPU limits, (i like memory limits) but CPU got throttled and it changes the behavior of the app. Removing CPU limit here

File: katalog/kube-proxy-exporter/deploy.yml (hunk 4 above)
Comment (angelbarrera92) [reply katalog/kube-proxy-exporter/deploy.yml:30]: Tested without hostPID and it continue working. Removed it :)

**Comment by angelbarrera92** [comment]
I'll remove the kong and glusterfs dashboards. If someone requires it, feel free to recover it from a previous release :)

**Pull request merged**